array.c

```diff
@@ -4466,29 +4410,21 @@ rb_ary_clear(VALUE ary)
 /*
  *  call-seq:
  *    array.fill(obj) -> self
- *    array.fill(obj, start) -> self
- *    array.fill(obj, start, length) -> self
+ *    array.fill(obj, start, length = self.size) -> self
```

Comment:
I don't think these cases are the same, so they should use a separate call-seq (as before), and documented separately.

```ruby
a = ['a', 'b', 'c', 'd']
a.dup.fill(:X, 2)
=> ["a", "b", :X, :X]
a.dup.fill(:X, 2, a.size)
=> ["a", "b", :X, :X, :X, :X]
```

Reply:
call-seq restored.  Do the examples need to change?